plugins/org.python.pydev/pysrc/_pydev_bundle/pydev_console_utils.py

```diff
@@ -153,7 +153,8 @@ def need_more_for_code(self, source):
         if hasattr(self.interpreter, 'is_complete'):
             return not self.interpreter.is_complete(source)
         try:
-            code = self.interpreter.compile(source, '<input>', 'exec')
+            symbol = 'single' if IS_JYTHON else 'exec'  # Jython does not support 'exec'
```

Comment:
Thanks for the patch ;)

A question: do you know if all Jython versions are the same here (i.e.: should the symbol be single only on some Jython versions?)

Also, can you change to not use the ternary operator? (the debugger can still work with pre 2.5 versions)